src/sinks/pulsar.rs

```diff
@@ -249,6 +268,12 @@ fn encode_event(mut item: Event, encoding: &EncodingConfig<Encoding>) -> crate::
             .get(log_schema().message_key())
             .map(|v| v.as_bytes().to_vec())
             .unwrap_or_default(),
+        Encoding::Avro => {
+            let schema = avro_rs::Schema::parse_str(encoding.schema().as_ref().unwrap()).unwrap();
```

Comment:
Generally we try to avoid `unwrap()` calls, since this function returns a `crate::Result` you should be able to use like this?

```suggestion
            let schema = avro_rs::Schema::parse_str(encoding.schema().as_ref()?)?;
```

Reply:
Can't apply it as you suggest.